docs/COMMIT.md

```diff
@@ -6,7 +6,23 @@ We agree on the following simple rules to make our lives easier :)
 - Stick to the **style** below for **commit messages**
 - **Commit compiling patches** for the *main* branches (`master` and `dev`),
   you can be less strict for (unshared) *topic branches*
-
+- Commits should be formated with clang-format-11
+
+Format Code
+-----------
+
+- Install clang-format-11
+- To format all file in your working copy, you can run this command in bash from the root folder of PIConGPU:
+  ```bash
+  find include/ share/picongpu/ share/pmacc -iname "*.def" \
+  -o -iname "*.h" -o -iname "*.cpp" -o -iname "*.cu" \
+  -o -iname "*.hpp" -o -iname "*.tpp" -o -iname "*.kernel" \
+  -o -iname "*.loader" -o -iname "*.param" -o -iname "*.unitless" \
+  | xargs clang-format-11 -i
+  ```
+ 
+To format only lines you added with `git add` you can call `git clang-format-11` before you create a commit.
+Take care that un-staged changes will not be formatted.
```

Comment:
Isn't this too strict and could scare off potential contributors who are not part of the core team?

Reply:
What do you mean, `Take care that un-staged changes will not be formatted.` is how git behaves if you try to format your changed code lines only. All that is not staged will not be formated by git.

Reply:
If you mean the formating itself. It is possible to write a bot that formats always code provided as a PR, but I think nobody of us knows how to do it, you need resources too where the bot is running or you need to integrate it somehow into the CI.
All that will not be ready in a short or mid time frame.

Reply:
> What do you mean, Take care that un-staged changes will not be formatted. is how git behaves if you try to format your changed code lines only. All code not staged will not be formated by git.

My comment was not about this sentense, but the whole highlighted part. I am just worried that this piece may seem too strict in a way that a new contributor who makes some small change (e.g. clarify comments or fix a typo) would be discouraged from doing so after reading this. E.g. a typical Windows user would not even have clang format.

A bot would of course be ideal, but I already assumed that would be too difficult to do quickly.

Reply:
Maybe I also misinterpret the tone and how strict it really is. @PrometheusPi @steindev @n01r what do you think?

Reply:
But yes everyone must format there changes else the CI will not pass see #3441 But we had this in the past too e.g. for end of line whitespaces and tabs, ...

Reply:
I think clang format is also available on windows or?

Reply:
Yes, it is. Just almost no one has it "by default"

Reply:
using clang format in visual studio https://devblogs.microsoft.com/cppblog/clangformat-support-in-visual-studio-2017-15-7-preview-1/

Reply:
That nobody has it can not solved by us or can only solved by a bot.

Reply:
I agree with @sbastrakov impression. The sentence `Take care that un-staged changes will not be formatted.` sound relatively strict. In the end most users should use git anyway. Thus perhaps one should write: `Please be aware that un-staged changes will not be formatted.` @sbastrakov Is this more the tone you wanted and @psychocoderHPC is this still the message you wanted to communicate?

Reply:
Ican update this sentense with the already open fix for the docu